### PR TITLE
Add a localization for counter text, separate into own semantic node

### LINKED ISSUE #19520: Talkback reads character counter as a fraction
In the trailing character counter on the TextField, "4/8"  is read as "four eighths".  Should probably be something like "4 out of 8" or "4 of 8"

### PR DESCRIPTION
**Pull request by jonahwilliams**
Based on a11y recommendation - the character counter should likely be exposed to accessibility as a separate focusable element distinct from the text entry field.

TalkBack will read "1/8" as a fraction, so we now provide a localized default.

Fixes https://github.com/flutter/flutter/issues/19520

### CODE REVIEW

**Comment by jonahwilliams** [comment]
Actually, this doesn't seem to quite work when the counter text is a direct child of the text field. Instead we'll most likely need to make them siblings.

**Comment by jonahwilliams** [comment]
The error noted above is unrelated to this, instead it is https://github.com/flutter/flutter/issues/21030

File: packages/flutter/lib/src/material/text_field.dart [hunk 1]
```diff
@@ -388,17 +403,25 @@ class _TextFieldState extends State<TextField> with AutomaticKeepAliveClientMixi
     if (!needsCounter)
       return effectiveDecoration;
 
-    final String counterText = '${_effectiveController.value.text.runes.length}/${widget.maxLength}';
+    final int currentLines = _effectiveController.value.text.runes.length;
+    final String counterText = '$currentLines/${widget.maxLength}';
```
Comment (goderbauer) [inline packages/flutter/lib/src/material/text_field.dart]: If talkback always reads this out as a fraction can we provide a better default as default? My guess is that most devs will forget to customize this....

**Review by goderbauer — APPROVED** [review]
LGTM

**Comment by jonahwilliams** [comment]
Adding @HansMuller for localization sanity check

File: packages/flutter/lib/src/material/input_decorator.dart [hunk 2]
```diff
@@ -2380,6 +2386,9 @@ class InputDecoration {
   /// This property is true by default.
   final bool enabled;
 
+  /// An alternative semantic label for the [counterText].
```
[the remark this answers is not in the document]
Comment (HansMuller) [reply packages/flutter/lib/src/material/input_decorator.dart]: Might help to answer "alternative to what?" here

File: packages/flutter_localizations/lib/src/material_localizations.dart [hunk 3]
```diff
@@ -393,6 +393,58 @@ abstract class GlobalMaterialLocalizations implements MaterialLocalizations {
...
   }
 
+  /// The "zero" form of [remainingTextFieldCharacterCount].
+  ///
+  /// This form is required.
+  ///
+  /// See also:
+  ///
+  ///  * [Intl.plural], to which this form is passed.
+  ///  * [selectedRowCountTitleZero], the "zero" form
```
Comment (HansMuller) [inline packages/flutter_localizations/lib/src/material_localizations.dart]: cut and paste problems here

File: packages/flutter/lib/src/material/text_field.dart [hunk 4]
```diff
@@ -388,17 +390,24 @@ class _TextFieldState extends State<TextField> with AutomaticKeepAliveClientMixi
     if (!needsCounter)
       return effectiveDecoration;
 
-    final String counterText = '${_effectiveController.value.text.runes.length}/${widget.maxLength}';
+    final int currentLength = _effectiveController.value.text.runes.length;
+    final String counterText = '$currentLength/${widget.maxLength}';
+    final int remaining = currentLength > widget.maxLength ? 0 : widget.maxLength - currentLength;
```
Comment (HansMuller) [inline packages/flutter/lib/src/material/text_field.dart]: or `(widget.maxLength - currentLength).clamp(0, widget.maxLength)`

**Review by HansMuller — APPROVED** [review]
LGTM

File: packages/flutter/lib/src/material/text_field.dart (hunk 4 above)
Comment (jonahwilliams) [reply packages/flutter/lib/src/material/text_field.dart]: Done

File: packages/flutter_localizations/lib/src/material_localizations.dart (hunk 3 above)
Comment (jonahwilliams) [reply packages/flutter_localizations/lib/src/material_localizations.dart]: Woops, fixed

File: packages/flutter/lib/src/material/input_decorator.dart (hunk 2 above)
[the remark this answers is not in the document]
Comment (jonahwilliams) [reply packages/flutter/lib/src/material/input_decorator.dart]: Fixed and linked from counterText

**Pull request merged**